### ISSUE TITLE
Scan freezes

### ISSUE THREAD

**Issue by Angi2412** [nexB/scancode-toolkit#2232]
### Description

I do have a setup, where ScanCode is used in a worker in combination with Redis an RQ. For some time now it was noticed, that sometimes the scan job just freezes. This shows in the CPU usage of the worker container suddenly goes from e.g. 100% to 0% and nothing happens until the container gets restarted. I tested it in various scenarios and it seems the "freezing" only happens during the ScanCode Scan (not in other scanner jobs e.g. CryptoDetector).
One example of a package where this freezing occurs is "[perl-Errno-1.28-416.el8.x86_64.rpm](https://centos.pkgs.org/8/okey-x86_64/perl-Errno-1.28-416.el8.x86_64.rpm.html)".

I use the following command for starting the scan:

```python
cores = multiprocessing.cpu_count()
rc, results = cli.run_scan(os.path.split(path)[0], license=True, copyright=True, return_results=True, processes=cores)
```

and the following to create the job:
```python
job = Job.create(hunt.hunt, args=(requestType, requestId, engine, temp_id), connection=r,
                             timeout=os.getenv("TIMEOUT"), id=sql.getHash(requestId, requestType, engine))
```

I attached screenshots of the log and resource usage while freezing form portainer:
log:
![log](https://user-images.githubusercontent.com/11177877/93340893-be692500-f82d-11ea-85c8-6790fb91817e.png)

ressouces:
![ressource](https://user-images.githubusercontent.com/11177877/93340899-bf01bb80-f82d-11ea-97a7-e7b414e79e49.png)

### How To Reproduce

Start a ScanCode scan for the package: "perl-Errno-1.28-416.el8.x86_64.rpm".

### System configuration

> For bug reports, it really helps us to know:

* What OS are you running on? (Windows/MacOS/Linux)
Linux
* What version of scancode-toolkit was used to generate the scan file?
3.2.0
* What installation method was used to install/run scancode? (pip/source download/other)
pip

**Comment by pombredanne** [comment]
@Angi2412 welcome back! 
Is your code public by chance? that would help with the debug. It is possible that since scancode uses multiprocessing and multithreading and that rq also uses multiprocessing that there is some unhappy conflict between the two.
1. can you try to reproduce this using the scancode command line directly?
2. based on your output there seems to be some debug logging turned on. May be a we can turn on more tracing on the scancode side. could you try to pass `verbose=True` and your own `echo_func` that would do logging? 
3. could you try to use `processes=-1` that would disable scancode own multiprocessing and threading
4. Is this consistently happening all the times when running the same scan?

**Comment by Angi2412** [comment]
@pombredanne Thanks for the fast response. At the moment the code is not public.
But it turns out this bug didn't occur for a whole week. I think the production machine didn't update ScanCode for some reason.
So i think this bug is resolved with 3.2.0 or the new version of RQ. But if it occurs again, I will reopen this or open a new issue. Thank you very much for the update :D
